### PR TITLE
Add command to install and configure Xdebug

### PR DESCRIPTION
**Pull request by lithrel**
## Install and configure

Configuration is in `.p4-env.json`, you can override it in your `.p4-env.override.json`
```json
{
  "planet4": {
    "xdebug": {
      "version": "3.3.1",
      "settings": {
        "xdebug.mode": "debug,profile,trace",
        "xdebug.start_with_request": "trigger",
        "xdebug.trigger_value": "StartXdebug",
        "xdebug.output_dir": "/var/www/html/wp-content",
        "xdebug.profiler_output_name": "profile.%u.%R",
        "xdebug.trace_output_name": "trace.%u.%R",
        "xdebug.trace_format": "3",
        "xdebug.discover_client_host": "true",
        "xdebug.client_port": "9003"
      }
    }
  }
}
```
With default configuration, traces and profiles will appear in your `./planet4` folder.
To install and configure, run:
```console
> npm run xdebug:install
```

If you modify your configuration, to re-configure xdebug run:
```console
> npm run xdebug:configure
```

XDebug will disappear from the container when it stops.

## Usage
Use a plugin like Xdebug helper in your browser ([Firefox](https://addons.mozilla.org/en-GB/firefox/addon/xdebug-helper-for-firefox/), [Chrome](https://chromewebstore.google.com/detail/xdebug-helper/eadndfjplgieldjbigjakmdgkmoaaaoc)), and set the trigger value in its settings. Enable Trace or Profile only when you want to generate a trace or profile file, and disable it when you're done.

## Profile
Use [KCacheGrind](https://kcachegrind.github.io/html/Home.html) or equivalent ([webgrind](https://github.com/jokkedk/webgrind), etc.) to open the `profile` files generated in your `./planet4` folder.
Screenshot from 2024-04-14 12-09-19

## Debug
Example of a `launch.json` configuration for VSCode (using [php debug](https://github.com/xdebug/vscode-php-debug) extension):
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "pathMappings": {
                "/var/www/html/wp-content": "${workspaceFolder}/planet4"
            },
            "log": true
        }
    ]
}
```

## Flamegraph
Run this command on a `trace` file to generate an interactive flamegraph. The SVG file will appear in your `./planet4` folder.

```console
> npm run xdebug:flamegraph planet4/trace.1713088263.986925._.xt.gz
```

trace 1713088263 986925 _ xt gz

### CODE REVIEW

File: .p4-env.json [hunk 1]
```diff
@@ -11,6 +11,20 @@
     },
     "composer": {
       "processTimeout": "600"
+    },
```
Comment (GP-Dan-Tovbein) [inline .p4-env.json:14]: Adding `"xdebug.client_host": "host.docker.internal"` to make it work the debugger on MacOs.

**Comment by GP-Dan-Tovbein** [comment]
Tested super deeper this PR and it finally worked as expected but with a couple of workarounds.

**Chip: Apple M2
MacOS Ventura 13.2.1**

### Debugger
✅ Worked fine after modifying these lines
```
{
  "planet4": {
    "xdebug": {
        ...,
        - "xdebug.discover_client_host": "true",
        + "xdebug.discover_client_host": "0",
        + "xdebug.client_host": "host.docker.internal"
      }
    }
  }
}
```

### Trace
✅ It's working fine, however somehow the generated files cannot be zipped.

When installing I get
```bash
checking whether to compress profiler files (requires zlib)... yes
checking for zlib >= 1.2.9... no
```
Screenshot 2024-05-13 at 12 48 16

## The solution
` /planet4/flamegraph/flamegraph.pl ./planet4/[trace_filename]._.xt > [svg_filename].svg`

## Extra information

### Firefox Add-on's setting
Screenshot 2024-05-13 at 12 32 52


Echoing `xdebug_info();`
Screenshot 2024-05-13 at 12 47 29

Screenshot 2024-05-13 at 12 47 40

**Review by GP-Dan-Tovbein — APPROVED** [review]
Tested locally with some workarounds and seems to be working perfect 👍 
Nice work @lithrel 👏

**Comment by lithrel** [comment]
- Integrated client_host configuration
- Fixed flamegraph command to run on non-gzipped files

**Comment by GP-Dan-Tovbein** [comment]
@lithrel I think we should update the commands in the [all p4 commands page](https://support.greenpeace.org/planet4/development/installation#all-commands).
What do you think?

**Pull request merged**